### ISSUE TITLE
GSE151091 - HumanMelanocyteDevelopment

### ISSUE THREAD

**Issue by idazucchi** [ebi-ait/hca-ebi-wrangler-central#853]
### Project short name:
HumanMelanocyteDevelopment
### Primary Wrangler:
Ami

### Secondary Wrangler:
Ida

### Associated files

* Google Drive:

https://drive.google.com/drive/folders/1I2owxOEI2f7_QMmRGp9gP4ijBuRt17RJ

**Merged FACS version:** **Validates in ingest (minus the data file transfer)**
https://docs.google.com/spreadsheets/d/1N-WgyxSkui1M_TAnuRAXiR6-in_fdoTXcAiPWMhGGfw/edit#gid=695411749

**Un-Merged FACS version V1:** **V1** **A:  experiment accessions in cell suspension tab**

https://docs.google.com/spreadsheets/d/1Ii8GHnrS7iGE3Tt4dS0XPkY4DxUpOn7l-gxKpGacIZc/edit#gid=1716876762

**Un-Merged FACS version V1:** **V1** **B:  no experiment accessions in cell suspension tab**

https://docs.google.com/spreadsheets/d/1sN_SpEL4279iFKJmQBolERYIG6I9uRvP40WQDoLHO_M/edit#gid=1716876762

**Un-Merged FACS version V2:** **V2** **A:  experiment accessions in cell suspension tab**
Extra cell suspensions to allow linking cell suspensions to analysis files. The total number of cell suspensions will not fit in a single cell in the spreadsheet, it reaches the maximum limit!

https://docs.google.com/spreadsheets/d/1Z5Ia8QyBP91L51TKMha_EKnj5KNPzJs6ZfbKux_yT8Q/edit#gid=1716876762

**Un-Merged FACS version V2:** **V2** **B:  No experiment accessions in cell suspension tab**
Extra cell suspensions to allow linking cell suspensions to analysis files. The total number of cell suspensions will not fit in a single cell in the spreadsheet, it reaches the maximum limit!

https://docs.google.com/spreadsheets/d/17c1X-390qVd3FdQxBoW4gQPeCf4xN7x7PQR4HvfIm6c/edit#gid=1701476767

**Un-Merged FACS version V3: - to be reviewed**

https://docs.google.com/spreadsheets/d/1eaSJjVSJPrAbdZD9JoMfqu8TOjC4uTKxbRrqCAO2PA0/edit#gid=1716876762

### Published study links

* Paper: [Human melanocyte development and melanoma dedifferentiation at single-cell resolution](https://doi.org/10.1038/s41556-021-00740-8)

* Accessioned data: [GSE151091](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE151091)

* Ingest: [project](https://contribute.data.humancellatlas.org/projects/detail?uuid=a4f154f8-5cc9-40b5-b8d7-af90afce8a8f&tab=project)

### Key Events

- [x] Convert published metadata to HCA spreadsheet
- [x] Manually curate dataset to meet HCA metadata standard
- [x] Collect any matrix and cell-type annotation files
- [x] Upload sheet to validate metadata
- [x] Check linking using ingest graph validator
- [x] Transfer raw files to ingest to validate data files
- [x] Ask the Secondary Wrangler for an end-to-end review of the project. Ask the Expertise Wrangler to review specific tabs if needed
- [ ] Submit dataset to Production 
- [ ] Complete the Export SOP
- [ ] Convert project data to SCEA format following the SCEA conversion SOP if appropriate

**Comment by ami-day** [comment]
Downloading fastq files.

**Comment by ami-day** [comment]
Need to change modelling to reflect individual wells as opposed to samples for smart-seq2.

**Comment by ami-day** [comment]
Some fastq files are not available in ENA.

**Comment by ami-day** [comment]
Requested SRA object files from NCBI Cloud Delivery.

**Comment by ami-day** [comment]
1a14f2aa-7b97-4910-8f3a-b67137bc1525

**Comment by ami-day** [comment]
Removed 3 runs from sequence file tab for which no data is available - tried obtaining fastq files and bam files and they are not available. Tried downloading the sra object files and an error is raised by the tool, they don't seem to be compatible with the tool. The missing runs are: SRR11671398, SRR11671399, SRR11671402

**Comment by ami-day** [comment]
Created test data for the experiment design (cell suspensions as input to cell suspension): https://staging.contribute.data.humancellatlas.org/submissions/detail?uuid=f8fe28ab-8286-41c6-afe7-4b66785e8480&project=6eee6cb1-dc3a-41e2-b8a9-280690f1ec0a

**Comment by ESapenaVentura** [comment]
@idazucchi to review when ready (test data being generated)

**Comment by ami-day** [comment]
Stuck in metadata validating status.

**Comment by idazucchi** [comment]
Hi Ami, I'm done with the secondary review! It looks good I just have minor comments

### Project
- the publication url points to the data availability section
- this is not an official HCA publication

### Donor
- Can you record the skin colour in the description?  The skin network told us they are very interested in this metadata
- you can add the enthicity

### Collection protocol
- the ontology label ended up in the retail name field

### Specimen
- the organ part field is not filled in. If you fill that in with the anatomical_location available from `metadata.csv` there should be 38 specimens
- you could improve the description by removing the numeric codes which are hard to interpret and add the donor name or the organ part

### Enrichment protocol
- there is a cell size selection step incorporated in the FACS protocol - can you add one more enrichment protocol for size selection 35µm ?
- is `enrichment_protocol_facs` needed? it doesn't select for any specific marker and it's not linked to any cell supsension
- it would be better to isolate the markers that are changing, so 1 protocol for 

### Cell suspension 
- you can add the estimated cell count = 1 for all smart-seq cell suspensions - not sure how to treat the SRX cell suspensions in this respect
- SRX cell suspensions:
  - the input cell suspensions are listed as input specimens instead
  - not sure of what protocols we should be using to link them together - do you want to discuss this?

### Library prep
- you could add the information about the spike in

### Sequence file
- some fastqs are not compressed - can you compress them?

### Analysis protocol
- I'm not sure we agreed on this but I think we are using `;` to separate the computational method field. We need to be consistent with the separator we use

### Analysis file
- the louvian files are not cell count matrices, the cell counts are not needed for them

**Comment by ami-day** [comment]
I cant edit the project as it's stuck in graph validating state

**Comment by ami-day** [comment]
Thanks @idazucchi I have made most of the suggested changes.

The enrichment_protocol_facs is needed for the experiment accessions linked to multiple cell_ids, as the cell_ids reflect different cell types enriched using different cell markers during facs.

I changed the relevant input specimens in the cell suspension tab to input cell suspensions (added this column), where the experiment accession is linked to multiple single cell ids (cell suspensions). I will need to update the test data I created for this scenario to reflect those changes.

I can't yet edit the Project, since it is in validating status, but will do when it's done.

**Comment by Wkt8** [comment]
Submission needs to be resubmitted

**Comment by ami-day** [comment]
Error deleting project.

**Comment by idazucchi** [comment]
[ops ticket](https://app.zenhub.com/workspaces/operations-5fa2d8f2df78bb000f7fb2b5/issues/ebi-ait/hca-ebi-wrangler-central/916)

**Comment by ami-day** [comment]
Test data still needs testing

**Comment by ami-day** [comment]
Getting an error when creating test data pull request:
https://app.zenhub.com/workspaces/operations-5fa2d8f2df78bb000f7fb2b5/issues/ebi-ait/hca-ebi-dev-team/506

**Comment by ami-day** [comment]
Fixed issue. Now waiting on pull request for the updated Schema_Test_Data.xlsx file in the Schema_Test_Data repo.

**Comment by ami-day** [comment]
Waiting on this PR: https://github.com/HumanCellAtlas/schema-test-data/pull/28

**Comment by Wkt8** [comment]
Unblocked by cell-to-cell-suspension test data.

**Comment by idazucchi** [comment]
more info on the gating / FACS selection in [this thread](https://groups.google.com/a/data.humancellatlas.org/g/wrangler-team/c/rq4aBvZTksA/m/ETKXb06QAAAJ)

**Comment by idazucchi** [comment]
The spreadsheet is ready for ingest but we are having trouble deleting the esxisting submission because of its size. 
the hca-util area is also ready `1a14f2aa-7b97-4910-8f3a-b67137bc1525`

**Comment by amnonkhen** [comment]
I see the [submission](https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/63ece9995cd8d87e76ebb1b5) was created on 15/2 (today). 
I don't understand the problem with this dataset. Which submission were you trying to delete @idazucchi ?

**Comment by idazucchi** [comment]
The submission is graph validating but the dataset is huge so I'll keep an eye out to see it it gets stuck
start time ~10AM

**Comment by idazucchi** [comment]
the graph validation failed - ticket to address this here ebi-ait/hca-ebi-wrangler-central#1050

**Comment by Wkt8** [comment]
Graph validation passed locally 

```
(ingest-graph-validator) wteh@C02DM256ML85 ~ % cd /Users/wteh/Documents/Wrangling/Published/Ami_latest_fake_project
(ingest-graph-validator) wteh@C02DM256ML85 Ami_latest_fake_project % ingest-graph-validator hydrate xls Ami_latest_fake_project.xlsx
23-02-21 16:08:43 [ingest_graph_validator.ingest_graph_validator] - INFO: starting neo4j docker instance
23-02-21 16:08:43 [ingest_graph_validator.ingest_graph_validator] - INFO: attached to backend container [neo4j-server]
23-02-21 16:08:44 [ingest_graph_validator.hydrators.hydrator] - INFO: started xls hydrator for file [Ami_latest_fake_project.xlsx]
23-02-21 16:09:29 [ingest_graph_validator.utils] - INFO: [import_spreadsheet] took [44952] ms
23-02-21 16:09:30 [ingest_graph_validator.hydrators.hydrator] - INFO: imported 19791 nodes
23-02-21 16:09:38 [ingest_graph_validator.hydrators.hydrator] - INFO: imported 253747 edges
23-02-21 16:21:32 [ingest_graph_validator.hydrators.hydrator] - INFO: hydration finished
(ingest-graph-validator) wteh@C02DM256ML85 Ami_latest_fake_project % ingest-graph-validator action test graph_test_set
23-02-21 17:00:09 [ingest_graph_validator.ingest_graph_validator] - INFO: starting neo4j docker instance
23-02-21 17:00:09 [ingest_graph_validator.ingest_graph_validator] - INFO: attached to backend container [neo4j-server]
Usage: ingest-graph-validator action test [OPTIONS] TEST_PATH
Try "ingest-graph-validator action test -h" for help.

Error: Invalid value for "TEST_PATH": Path "graph_test_set" does not exist.
(ingest-graph-validator) wteh@C02DM256ML85 Ami_latest_fake_project % ingest-graph-validator action test /Users/wteh/PycharmProjects/ingest-graph-validator/graph_test_set
23-02-21 17:00:18 [ingest_graph_validator.ingest_graph_validator] - INFO: starting neo4j docker instance
23-02-21 17:00:18 [ingest_graph_validator.ingest_graph_validator] - INFO: attached to backend container [neo4j-server]
23-02-21 17:00:18 [ingest_graph_validator.actions.test_action] - INFO: loading tests
23-02-21 17:00:18 [ingest_graph_validator.actions.test_action] - INFO: loaded [16] test queries
23-02-21 17:00:18 [ingest_graph_validator.actions.test_action] - INFO: running tests
23-02-21 17:00:20 [ingest_graph_validator.actions.test_action] - INFO: all tests finished 
(ingest-graph-validator) wteh@C02DM256ML85 Ami_latest_fake_project % 





```

**Comment by idazucchi** [comment]
Project is graph valid - currently waiting for ebi-ait/dcp-ingest-central#926 to be in production before trying to export
This project has failed spreadsheet generation due to the large linkingMap in a previous attempt

**Comment by Wkt8** [comment]
Stuck in submitted. -needs exporting properly.

**Comment by arschat** [comment]
successfully exported (described in #1082) and input form sent.
